test/functional/test_framework/p2p.py

```diff
@@ -539,6 +566,33 @@ def close(self, timeout=10):
         # Safe to remove event loop.
         NetworkThread.network_event_loop = None
 
+    @classmethod
+    def listen(cls, p2p, callback, port=None, addr=None, idx=1):
+        if port is None:
+            assert 0 < idx <= MAX_NODES
+            port = p2p_port(MAX_NODES - idx)
+        if addr is None:
+            addr = '127.0.0.1'
+        coroutine = cls.create_server(addr, port, callback, p2p)
+        cls.network_event_loop.call_soon_threadsafe(cls.network_event_loop.create_task, coroutine)
+
+    @classmethod
+    def peer_listener(cls, addr, port):
+        def pl():
+            x = cls.protos.get((addr,port))
+            cls.protos[(addr,port)] = None
+            return x
+        return pl
+
+    @classmethod
+    async def create_server(cls, addr, port, callback, proto):
```

Comment:
nit: I'm really not fond of this name, being that it appears to be an intentional collision with `asyncio`'s `create_server`.  It's hard enough to tell _who's_ calling _what_ and _when_ in concurrent python, and this makes it harder.  Maybe something like `listen_helper`?

Reply:
yeah that's reasonable, I updated to `create_listen_server`